### PR TITLE
[DROOLS-6549] remove v7 rule units

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6549

Upstream PR: https://github.com/kiegroup/droolsjbpm-knowledge/pull/573

### CODE REVIEW

File: drools-scenario-simulation/drools-scenario-simulation-backend/pom.xml [hunk 1]
```diff
@@ -47,6 +47,10 @@
       <groupId>org.drools</groupId>
       <artifactId>drools-core</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.drools</groupId>
+      <artifactId>drools-mvel</artifactId>
```
Comment (danielezonca) [inline drools-scenario-simulation/drools-scenario-simulation-backend/pom.xml:52]: Why this change? Is it related to rule unit PR?

**Review by danielezonca — COMMENTED** [review]
The ticket describes to review also the class `StatefulKnowledgeSessionImpl` to remove changes implemented only to make old RuleUnit work (`bindRuleBase` and in general the possibility to create a KieSession without a KieBase)

**Review by tarilabs — COMMENTED** [review]
I'm a bit surprised this works without any change to Tests.
To be clear, I would expect this of course no longer works with `generateDMNmodel` in kie-maven-plugin testing, and which is fine.

I'm a bit confused the test does not break for executing with Rule Unit v7 in -core for DMN.

Did you modify any tests and I missed it please?

**Comment by tarilabs** [comment]
copied @lucamolteni for the changes even if marginally would impact ANC

File: drools-scenario-simulation/drools-scenario-simulation-backend/pom.xml (hunk 1 above)
Comment (mariofusco) [reply drools-scenario-simulation/drools-scenario-simulation-backend/pom.xml:52]: This module depends on `drools-mvel` and before it took this dependency transitively through `dmn-core` and `drools-ruleunit`

```
[INFO] +- org.kie:kie-dmn-core:jar:8.13.0-SNAPSHOT:compile
[INFO] |  +- org.drools:drools-ruleunit:jar:8.13.0-SNAPSHOT:compile
[INFO] |  |  \- org.drools:drools-mvel:jar:8.13.0-SNAPSHOT:compile

```

**Comment by lucamolteni** [comment]
> I'm a bit surprised this works without any change to Tests. To be clear, I would expect this of course no longer works with `generateDMNmodel` in kie-maven-plugin testing, and which is fine.
> 
> I'm a bit confused the test does not break for executing with Rule Unit v7 in -core for DMN.
> 
> Did you modify any tests and I missed it please?

I think it's running test in the non-exec mode two times. 

I've already told @mariofusco to fix the tests

**Comment by tarilabs** [comment]
> I've already told @mariofusco to fix the tests

Or I can work personally on this, I'm just trying to understand what this PR does and what does not do, so to eventually adapt the rest in separate PRs.

Let me know!

**Comment by mariofusco** [comment]
@lucamolteni was right, it was testing the same thing twice. @tarilabs I agreed with him to keep the tests parametrized because this will be necessary when the compiled alpha network will be finished, but to make them run only with the exec model parameter for now.

**Review by danielezonca — APPROVED** [review]

**Review by tarilabs — APPROVED** [review]
thank you @mariofusco @lucamolteni

**Review by lucamolteni — APPROVED** [review]

**Comment by mariofusco** [comment]
jenkins retest this

**Comment by mariofusco** [comment]
jenkins retest this

File: kie-dmn/kie-dmn-core/pom.xml [hunk 2]
```diff
@@ -243,6 +239,11 @@
       <artifactId>jackson-datatype-jsr310</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>jakarta.xml.bind</groupId>
+      <artifactId>jakarta.xml.bind-api</artifactId>
+      <scope>provided</scope>
+    </dependency>
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-core/pom.xml:246]: how about doing like it's done in all the pom.xml we've learned this morning:
```suggestion
    <!-- prepare for JDK11 -->
    <dependency>
      <groupId>com.sun.xml.bind</groupId>
      <artifactId>jaxb-xjc</artifactId>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>com.sun.xml.bind</groupId>
      <artifactId>jaxb-core</artifactId>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>com.sun.xml.bind</groupId>
      <artifactId>jaxb-impl</artifactId>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>jakarta.xml.bind</groupId>
      <artifactId>jakarta.xml.bind-api</artifactId>
      <scope>provided</scope>
    </dependency>
```

Comment (tarilabs) [reply kie-dmn/kie-dmn-core/pom.xml:246]: you can directly push the button^

**Comment by gitgabrio** [comment]
@danielezonca @mariofusco 
Is `kie-maven-plugin` invoked from this build ? If I'm not wrong this PR is specific for DROOLS 8.x, and droolsjbpm-integration should not have relationship with that...

**Comment by danielezonca** [comment]
> @danielezonca @mariofusco Is `kie-maven-plugin` invoked from this build ? If I'm not wrong this PR is specific for DROOLS 8.x, and droolsjbpm-integration should not have relationship with that...

Compile downstream build is not "fixed" yet (in 8.x branch), it should not start or at least it should not reach -integration repo.

**Comment by mariofusco** [comment]
jenkins retest this

File: kie-dmn/kie-dmn-core/pom.xml (hunk 2 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-core/pom.xml:246]: discussed with MArio on chat, if only 1 provided dependency is necessary, it is fine to have only 1 additional provided dependency and not necessarily these 4.
